### ISSUE TITLE
proposal: additional wrapper types

### ISSUE THREAD

**Issue by bszonye** [sumpfork/dominiontabs#371]
The `--wrapper` option creates a specific kind of slipcase that is popular in the Dominion community. My own storage solution doesn't have room for pop-up tabs, so I've been using a different kind of wrapper that is more like a matchbook cover. I've also been experimenting with some pull-tab / ribbon designs. To support this, I've been working on a set of new wrapper options to generalize the different possible combinations.

The overall design splits the dividers into three main panels: card, head, and tail. The card is just a rectangle that works like the current `--front` with an optional back for two-sided printing.

The tail attaches to the bottom of the main card. It generalizes the current tab-folder style of the `--wrapper` feature, so that you can customize the length of the folder, remove the tab, or make the whole thing tab-width instead of card-width.

The head is everything that can be attached to the top of the main card: an index tab, a longer pull tab, or a matchbox-style flap. The head also has a "spine" subpanel to customize the flat bit of wrapper that goes over the top of the cards (the place where the current `--wrapper` puts a small copy of the card name). The spine can show the card name, the card type, or the full banner that goes on tabs.

Here are all of the new options from my work in progress:
```
Card Sleeves/Wrappers:
  Generating dividers that are card sleeves/wrappers.

  --wrapper             Draw sleeves (wrappers) instead of dividers for the
                        cards. Same as --head=strap --tail=folder (default:
                        False)
  --pull-tab            Draw folding pull tabs instead of dividers for the
                        cards. Same as --head=tab --tail=strap (default:
                        False)
  --cover               Draw folding covers instead of dividers for the cards.
                        Same as --head=wrapper --head-facing=back --head-
                        text=back (default: False)
  --head {tab,strap,cover,folder,none}
                        Top tab or wrapper type; 'tab' will add divider tabs;
                        'strap' will add a longer folding tab; 'cover' will
                        create a matchbook-style cover; 'none' will leave the
                        top edge plain. (default: tab)
  --tail {tab,strap,cover,folder,none}
                        Bottom wrapper type; 'strap' will add a folding pull
                        tab; 'cover' will create a simple folding cover;
                        'folder' will create a tab folder; 'none' will leave
                        the bottom edge plain. (default: none)
  --head-facing {front,back}
                        Text facing for top tabs and wrappers; 'front' will
                        show the text upright when flat or unfolded; 'back'
                        will show it upright when folded over. (default:
                        front)
  --tail-facing {front,back}
                        Text facing for tail wrappers; 'front' will show the
                        text upright when flat or unfolded; 'back' will show
                        it upright when folded over. (default: back)
  --head-text {card,rules,blank,front,back}
                        Text to print on top wrappers; 'card' will print the
                        text from the game card; 'rules' will print additional
                        rules for the game card; 'blank' will not print text
                        on the divider; 'front' will use the same setting as
                        --front; 'back' will use the same setting as --back.
                        (default: blank)
  --tail-text {card,rules,blank,front,back}
                        Text to print on tail wrappers; 'card' will print the
                        text from the game card; 'rules' will print additional
                        rules for the game card; 'blank' will not print text
                        on the divider; 'front' will use the same setting as
                        --front; 'back' will use the same setting as --back.
                        (default: back)
  --head-height HEAD_HEIGHT
                        Height of the divider head in centimeters (a value of
                        0 uses tab height or card height as appropriate).
                        (default: 0.0)
  --tail-height TAIL_HEIGHT
                        Height of the tail wrapper in centimeters (a value of
                        0 uses tab height or card height as appropriate).
                        (default: 0.0)
  --spine {name,types,tab}
                        Text to print on the top edge of wrappers; 'name' will
                        print the card name. 'type' will print the card
                        type. 'tab' will print all tab text and icons.
                        (default: name)
```

**Comment by bszonye** [comment]
Still working on this. I've implemented some major improvements to drawOutline, drawTab, and drawCost to make them more general, match the card layouts better, and streamline the code. For example, the new drawOutline doesn't have to handle wrappers as a special case anymore, and a lot of other special cases just become parameters.

The last major piece is to update drawText to handle the new panel and orientation options, and then I'll send a PR for review. I was going along at a pretty good pace until some personal stuff came up. Hopefully I'll be able to get back at it within a few days.

**Comment by sumpfork** [comment]
Cool, @wvoigt might be the right person to review if they have time, otherwise I will.

**Comment by bszonye** [comment]
Oops, turns out the personal stuff was pretty heavy, and I didn't get back to the project until today. I've just updated my workspace with all of the upstream changes. I need to refamiliarize myself with the project after so long away! Still hoping that it doesn't take too much longer to get a PR together.

**Comment by bszonye** [comment]
Back to actively working on this again.

**Comment by bszonye** [comment]
Resolved in #433.
